python/lsst/pipe/tasks/objectMasks.py

```diff
@@ -104,24 +114,52 @@ def readFits(fileName, hdu=0, flags=0):
-                mat = re.search(r"^\s*circle(?:\s+|\s*\(\s*)"
+                mat = re.search(r"^\s*(box|circle)(?:\s+|\s*\(\s*)"
                                 "(\d+(?:\.\d*)([d]*))" "(?:\s+|\s*,\s*)"
-                                "([+-]?\d+(?:\.\d*)([d]*))" "(?:\s+|\s*,\s*)"
-                                "(\d+(?:\.\d*))([d'\"]*)" "(?:\s*|\s*\)\s*)"
-                                "\s*#\s*ID:\s*(\d+)" "\s*$", line)
+                                "([+-]?\d+(?:\.\d*))([d]*)" "(?:\s+|\s*,\s*)"
+                                "([+-]?\d+(?:\.\d*))([d]*)" "(?:\s+|\s*,\s*)?"
+                                "(?:(\d+(?:\.\d*))([d'\"]*)\s*,\s*(\d+)(?:\.\d*([d]*))?)?"
+                                "(?:\s*|\s*\)\s*)"   # close paren
+                                "\s*#\s*ID:\s*(\d+)" # start comment
+                                "(?:\s*,\s*mag:\s*(\d+\.\d*))?"
+                                "\s*$", line)
                 if mat:
-                    ra, raUnit, dec, decUnit, radius, radiusUnit, _id = mat.groups()
+                    _type, ra, raUnit, dec, decUnit, \
+                        param1, param1Unit, param2, param2Unit, param3, param3Unit, \
+                        _id, mag = mat.groups()
 
                     _id = int(_id)
+                    if mag is None:
+                        mag = "NaN"
+                        
+                    mag = float(mag)
+
                     ra = convertToAngle(ra, raUnit, "ra", fileName, lineNo)
                     dec = convertToAngle(dec, decUnit, "dec", fileName, lineNo)
-                    radius = convertToAngle(radius, radiusUnit, "radius", fileName, lineNo)
+
+                    if _type == "box":
+                        width = convertToAngle(param1, param1Unit, "xsize", fileName, lineNo)
+                        height = convertToAngle(param2, param2Unit, "ysize", fileName, lineNo)
+                        angle = convertToAngle(param3, param3Unit, "theta", fileName, lineNo)
+
+                        if angle != 0.0:
+                            log.warn("Rotated boxes are not supported: \"%s\" at %s:%d" % (
+                                line, fileName, lineNo))                            
+                    elif _type == "circle":
+                        radius = convertToAngle(param1, param1Unit, "radius", fileName, lineNo)
 
                     rec = brightObjects.addNew()
                     # N.b. rec["coord"] = Coord is not supported, so we have to use the setter
+                    rec["type"] = _type
                     rec["id"] = _id
+                    rec["mag"] = mag
                     rec.setCoord(afwCoord.Fk5Coord(ra, dec))
-                    rec["radius"] = radius
+
+                    if _type == "box":
+                        rec["height"] = height
+                        rec["width"] = width
+                    elif _type == "circle":
+                        rec["radius"] = radius
```

Comment:
Experimentally afwTable does that, but I've made it explicit.  Removes a second if block looking
at region types too.